srcStatic/WindowsUniqueModule.h

```diff
@@ -0,0 +1,18 @@
+#include <windows.h>
+#include <memory>
+
+
+// Sentinel value guarded HMODULE deleter
+// For use with LoadLibrary (will automatically call FreeLibrary)
+struct ModuleDeleter {
+	using pointer = HMODULE;
+
+	void operator()(HMODULE handle) const {
+		if (handle != NULL) {
```

Comment:
Should this be nullptr instead of NULL. Or maybe it could be 

```C++
if (handle) {
    ...
}
```

Reply:
Hmm, interesting point. I do like that simplification. I suppose technically correctness depends on the definition of `NULL`, which we aren't in control of. Though realistically, the definition of `NULL` isn't about to change.

I checked real quick what the zero-as-null-pointer warning flag has to say about it. No warning is produced.

I think I'm tempted to leave this given the definition of `NULL` is external.

Reply:
Could you expand on your thought process? I'm not following why we would not use the keyword definition of nullptr to check if a handle is null (I think of a handle as a pointer).

-Brett

Reply:
A `HANDLE` is currently implemented as a pointer, though it could also be implemented as a numeric integer value. It's an opaque type. You don't really know how it's implemented, and the internal details could change. They probably won't, but conceivably a future (or past) version of Windows could choose to use a different internal representation. They would also be free to change the sentinel `NULL` value. Instead of it being `0` or `nullptr`, it could become some other magic number, such as `-1`. If the definition of `NULL` changed, and we weren't using `NULL` itself, the code would no longer be correct.

Reply:
Interesting, that article talks about wrapping handles like you are doing here.